src/aerovaldb/utils/encode.py

```diff
@@ -0,0 +1,33 @@
+def encode_str(string: str, *, encode_chars: dict[str, str]):
+    ls: list[str] = []
+    prev = 0
+    i = 0
+    while i < len(string):
+        if string[i] in encode_chars:
+            ls.append(string[prev:i] + encode_chars.get(string[i]))  # type: ignore
+            prev = i + 1
+        i += 1
+
+    ls.append(string[prev:])
+
+    return "".join(ls)
+
+
+def decode_str(string: str, *, encode_chars: dict[str, str]):
+    ls: list[str] = []
+    prev = 0
+    i = 0
+    while i < len(string):
+        if string[i] != "%":
+            i += 1
```

Comment:
Undocumented function, which might be the opposite from `encode_str`, but why does this function have a hardcoded "%", while `encode_str` doesn't?